p4-16/psa/psa.p4

```diff
@@ -568,12 +579,13 @@ control ComputeChecksum<H, M>(inout H hdr, inout M user_meta);
 
 control Deparser<H>(packet_out buffer, in H hdr);
 
-package PSA_Switch<H, M>(Parser<H, M> p,
-                         VerifyChecksum<H, M> vr,
+package PSA_Switch<H, M>(IngressParser<H, M> ip,
                          Ingress<H, M> ig,
+                         Deparser<H> igdep,
+                         EgressParser<H, M> ep,
                          Egress<H, M> eg,
                          ComputeChecksum<H, M> ck,
-                         Deparser<H> dep);
+                         Deparser<H> egdep);
```

Comment:
My point was that if the programmer decides to put checksum calculation anywhere but at the very end of the ingress/egress control, then we will have a problem. Checking for "the very end" whatever it might mean will be painful and we do not want to express it in the language.

I think it is much easier to say (and check) the rule "this extern should be used in that control only".

I think you are making an important point that recalculating checksums in the deparser requires the headers to become inout. Perhaps, that's a sign that we should have it separate and keep the deparser clean (i.e. do not allow it to modify headers) in that regard?